### DISCUSSION TITLE
websocket

### DISCUSSION THREAD

**Discussion by PP-HashInclude** [deta/cloud-docs#347] [General] [answered]
Hello,
I am getting error WebSocket connection to 'wss://xxxxxxxxxx.deta.dev/ws' failed: when I try to connect to websocket endpoint from javascript created using fastapi.  Appreciate any pointers to resolve the error.  I am able to load the default page https://xxxxxxxxxx.deta.dev/. Thanks.

**Comment by lemonyte** [comment, marked as answer]
WebSockets aren't supported by Deta because Micros are basically serverless functions that get killed after a 10 second timeout, so long-running connections are impossible.
